File: 算法/C++/507. Perfect Number.cpp
class Solution {
public:
    bool checkPerfectNumber(int num) {
        int c = pow(num, 0.5) + 1;
        set<int> factors;
        if (num <= 3)
            return false;
        for (int i = 2; i <= c; i++) {
            if (num % i == 0)
            {
                factors.insert(i);
                factors.insert(num / i);
            }
        }
        int sum = 1;
        for (auto &it : factors)
        {
            sum += it;
        }
        if (sum == num)
            return true;
        return false;
    }
};

File: 算法/Python/507. Perfect Number.py
class Solution:
    def checkPerfectNumber(self, num):
        """
        :type num: int
        :rtype: bool
        """

        if num <= 3:
            return False

        factors = set()
        for i in range(2, int(num**0.5)+1):
            if num % i == 0:
                factors.add(i)
                factors.add(num//i)

        factors = list(factors)
        if sum(factors) == num-1:
            return True
        return False
